### PR TITLE
Add GraphQL interface

### LINKED ISSUE #2: GraphQL
Currently this project exposes only traditional RESTful API endpoints, but this interface is not comfortable until we make a basic front-end at least.  However, we currently don't have enough time to build it right now.

So I suggest to use GraphQL instead, because there are already existing IDEs for GraphQL.  We could gain time by using these IDEs as a minimum front-end.

### PR DESCRIPTION
**Pull request by jckdotim**
# Introduction

This PR adds `/graphql/` endpoint, which can request data by using [GraphQL] query.

Closes #2 

# How to use

```
curl 'http://localhost:5000/graphql?query=\{blocks\{index,nonce,miner,previousHash,transactions\{signer,publicKey,updatedAddress,timestamp,signature,actions\{plainValue\{key,value\}\}\}\}\}'
```

[GraphQL]: https://graphql.org/

### CODE REVIEW

File: Libplanet.Explorer/Controllers/ExplorerController.cs [hunk 1]
```diff
@@ -36,6 +39,129 @@ public BlockChain<T> GetBlockChain()
             return chain;
         }
 
+        public class BlockType : ObjectGraphType<Block<T>>
```
Comment (dahlia) [inline Libplanet.Explorer/Controllers/ExplorerController.cs]: Could you move these object graph types to a separated namespace, like `Libplanet.Explorer.GraphTypes`?

File: Libplanet.Explorer/Controllers/ExplorerController.cs [hunk 2]
```diff
@@ -36,6 +39,129 @@ public BlockChain<T> GetBlockChain()
...
+                        ulong index = context.GetArgument<ulong>("index");
+                        return _chain.FirstOrDefault(x => (ulong)x.Index == index);
+                    }
+                );
+                _chain = chain;
+            }
+        }
+        
+        [HttpGet("/graphql/")]
+        public IActionResult getGraphQLResult(
```
Comment (dahlia) [inline Libplanet.Explorer/Controllers/ExplorerController.cs]: Although existing code sticks with this Java-style convention, could you rename it to `GetGraphQLResult`?  (We should rename other existing names too.)

**Review by dahlia — COMMENTED** [review]
We are avoiding cross-merge commits (i.e., that a feature branch merges the master branch, and then the master branch merges back that feature branch again).  Could you remove a merge commit 55810e7a06ac45424620debe6880c37a6e1a7142 and rebase the whole commits on the current master?

It currently looks like:

```
* 659cde6 - (HEAD -> issue-2, jc/issue-2) Support IAction type <JC Kim>
*   55810e7 - Merge branch 'master' of github.com:planetarium/libplanet-explorer into issue-2 <JC Kim>
|\  
| *   18c1339 - Merge pull request #5 from longfin/feature/blocks-details <Swen Mun>
| |\  
| | * b9b22b9 - Check chain ID existence <longfin>
| | * 8c49684 - Add miner and difficulty to /blocks/ <longfin>
| |/  
| *   d5ada49 - Merge pull request #4 from dahlia/host-port-options <Hong Minhee>
| |\  
| | * 3bf3d8f - -H/--host and -p/--port options <Hong Minhee>
| | * fe63331 - Remove debug print <Hong Minhee>
| |/  
| *   b3d653e - Merge pull request #3 from dahlia/master <Hong Minhee>
| |\  
| | * 3730935 - (origin/master) Build for pull requests <Hong Minhee>
| | * e5deec7 - Upload built binaries <Hong Minhee>
| | * 65ee6e1 - Build & pack using GitHub Actions <Hong Minhee>
| |/  
* | aa6f010 - Add PATH argument info to "How to run" section <JC Kim>
* | f62260b - Add TransactionType <JC Kim>
* | 4b7959b - Add BlockType, BlockQuery, `/graphql/` endpoint <JC Kim>
* | eef6742 - Add GraphQL dependency <JC Kim>
|/  
* a8edbca - (jc/master) Merge pull request #1 from dahlia/exe <Hong Minhee>
```

but it instead should be like:

```
* 659cde6 - (HEAD -> issue-2) Support IAction type <JC Kim>
* aa6f010 - Add PATH argument info to "How to run" section <JC Kim>
* f62260b - Add TransactionType <JC Kim>
* 4b7959b - Add BlockType, BlockQuery, `/graphql/` endpoint <JC Kim>
* eef6742 - Add GraphQL dependency <JC Kim>
* ee453d5 - (upstream/master, master) Merge pull request #7 from dahlia/auto-reloader <Hong Minhee>
```

I must be possible by simply running the following command:

```bash
git checkout issue-2
git pull --rebase git://github.com/planetarium/libplanet-explorer.git
```

**Comment by jckdotim** [comment]
@dahlia PTAL

**Review by dahlia — APPROVED** [review]

**Comment by dahlia** [comment]
Thanks for your contribution!

**Pull request merged**